qcodes/instrument/parameter.py

```diff
@@ -1349,35 +1381,89 @@ def _update_with(self, *,
         def __call__(self) -> ParamDataType:
             return self.get(get_if_invalid=True)
 
-    def __init__(self, name: str, source: Parameter, *args: Any,
+    def __init__(self, name: str, source: Optional[Parameter], *args: Any,
                  **kwargs: Any):
-        self.source = source
 
-        for ka, param in zip(('unit', 'label', 'snapshot_value'),
-                             ('unit', 'label', '_snapshot_value')):
-            kwargs[ka] = kwargs.get(ka, getattr(self.source, param))
+        self._attr_inherit = {"label": {"fixed": False,
+                                        "default_none_value": name},
+                              "unit": {"fixed": False,
+                                       "default_none_value": ""}}
+
+        for attr, attr_props in self._attr_inherit.items():
+            if attr in kwargs:
+                attr_props["fixed"] = True
+            else:
+                attr_props["fixed"] = True
+            source_attr = getattr(source, attr,
+                                  attr_props["default_none_value"])
+            kwargs[attr] = kwargs.get(attr, source_attr)
 
         for cmd in ('set_cmd', 'get_cmd'):
             if cmd in kwargs:
                 raise KeyError(f'It is not allowed to set "{cmd}" of a '
                                f'DelegateParameter because the one of the '
                                f'source parameter is supposed to be used.')
+        if source is None and ("initial_cache_value" in kwargs
+                               or "initial_value" in kwargs):
+            raise KeyError("It is not allowed to supply 'initial_value'"
+                           " or 'initial_cache_value' "
+                           "without a source.")
+
         initial_cache_value = kwargs.pop("initial_cache_value", None)
```

Comment:
this is not related to this PR but i wonder why we don't do the same for the initial_value...

Reply:
Because initial_value will call set on the parameter which wil call set on the underlying parameter which the cache can read from. It is all covered in tests :) However inital_cache_value will only set the value in the `_Cache` that is replaced with the `DelegateCache` below

Reply:
yes, make sense, thanks!